src/internal/DotStarEsp32DmaSpiMethod.h

```diff
@@ -0,0 +1,320 @@
+    }
+
+    bool IsReadyToUpdate() const
+    {
+        spi_transaction_t t;
+        spi_transaction_t * tptr = &t;
+        esp_err_t ret = spi_device_get_trans_result(_spiHandle, &tptr, 0);
+
+        // We know the previous transaction completed if we got ESP_OK, and we know there's no transactions queued if tptr is unmodified
+        return (ret==ESP_OK || tptr == &t);
+    }
+
+    void Initialize(int8_t sck, int8_t dat0, int8_t dat1, int8_t dat2, int8_t dat3, int8_t ss)
+    {
+        memset(_data, 0x00, _sizeStartFrame);
+        memset(_data + _sizeStartFrame + _sizePixelData, 0x00, _spiBufferSize - (_sizeStartFrame + _sizePixelData));
+
+        _ssPin = ss;
+
+        esp_err_t ret;
+        spi_bus_config_t buscfg;
+        memset(&buscfg, 0x00, sizeof(buscfg));
+
+        buscfg.miso_io_num=dat1;
+        buscfg.mosi_io_num=dat0;
+        buscfg.sclk_io_num=sck;
+        buscfg.quadwp_io_num=dat2;
+        buscfg.quadhd_io_num=dat3;
+        buscfg.max_transfer_sz=_spiBufferSize;
+
+        //Initialize the SPI bus
+        ret=spi_bus_initialize(T_SPIBUS::spiHostDevice, &buscfg, T_SPIBUS::dmaChannel);
+        ESP_ERROR_CHECK(ret);
+
+        InitSpiDevice();
+    }
+
+    void Initialize(int8_t sck, int8_t miso, int8_t mosi, int8_t ss)
+    {
+        Initialize(sck, mosi, miso, -1, -1, ss);
+    }
+
+    // If pins aren't specified, initialize bus with just the default SCK and MOSI pins for the SPI peripheral (no SS, no >1-bit pins)
+    void Initialize()
+    {
+        if(T_SPIBUS::spiHostDevice == VSPI_HOST) {
+            Initialize(SCK, -1, MOSI, -1, -1, -1);
+        } else {
+            Initialize(14, -1, 13, -1, -1, -1);
+        }
+    }
+
+    void Update(bool)
+    {
+        while(!IsReadyToUpdate());
+
+        memcpy(_dmadata, _data, _spiBufferSize);
+
+        memset(&_spiTransaction, 0, sizeof(spi_transaction_t));
+        _spiTransaction.length=(_spiBufferSize) * 8; // in bits not bytes!
+        if(T_SPIBUS::parallelBits == 1)
+            _spiTransaction.flags = 0;
+        if(T_SPIBUS::parallelBits == 2)
+            _spiTransaction.flags = SPI_TRANS_MODE_DIO;
+        if(T_SPIBUS::parallelBits == 4)
+            _spiTransaction.flags = SPI_TRANS_MODE_QIO;
+        _spiTransaction.tx_buffer = _dmadata;
+
+        esp_err_t ret = spi_device_queue_trans(_spiHandle, &_spiTransaction, 0);  //Transmit!
+        assert(ret==ESP_OK);            //Should have had no issues.
+    }
+
+    uint8_t* getData() const
+    {
+        return _data + _sizeStartFrame;
+    };
+
+    size_t getDataSize() const
+    {
+        return _sizePixelData;
+    };
+
+    void applySettings(const SettingsObject& settings)
+    {
+        _speed.applySettings(settings);
+        if(_spiHandle) {
+            DeInitSpiDevice();
+            InitSpiDevice();
+        }
+    }
+
+private:
+    void InitSpiDevice()
+    {
+        spi_device_interface_config_t devcfg;
+        memset(&devcfg, 0x00, sizeof(devcfg));
```

Comment:
you should be able to replace the memset with an initializer.
```
spi_device_interface_config_t devcg = {0};
```

Reply:
I wasn't familiar with this technique and tried to find a reference.  I couldn't find a reference that said to include 0 (or another value) in the braces, but this reference says to just use empty braces.

https://en.cppreference.com/w/cpp/language/value_initialization

 Is this ok?
 
`spi_device_interface_config_t devcfg = {};`

Reply:
sure; do a quick test after it to make sure it is truly all zeros (debug snippet of code that you remove).  It is safer than memset zero if the best default to some internal object isn't zero and they implemented a default constructor for it.